### ISSUE TITLE
there is a difference between the test and validation results

### ISSUE THREAD

**Issue by Dev-HJYoo** [Pointcept/Pointcept#533]
Hello, thank you for your great work. 

I'm currently training pt-v3 models on the custom dataset. 

i notice the problem: **the difference between the test and validation results**

config.py

```python
_base_ = ["../_base_/default_runtime.py"]

# misc custom setting
batch_size = 24  # bs: total bs in all gpus
mix_prob = 0.8
empty_cache = False
enable_amp = True

# # fc layer unload
# hooks = [dict(
#     type="CheckpointLoader", 
#     keywords="backbone.final", 
#     replacement="backbone.unused"
#     ),
# ]

freeze = False


# model settings
model = dict(
    type="DefaultSegmentor",
    backbone=dict(
        type="SpUNet-v1m1",
        in_channels=4,
        num_classes=12,
        channels=(32, 64, 128, 256, 256, 128, 96, 96),
        layers=(2, 3, 4, 6, 2, 2, 2, 2),
    ),
    criteria=[
        dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=2), # sky
        dict(type="LovaszLoss", mode="multiclass", loss_weight=1.0, ignore_index=2), # sky
    ],
)

# scheduler settings
epoch = 1
eval_epoch = 1
optimizer = dict(type="AdamW", lr=0.02, weight_decay=0.005)
scheduler = dict(
    type="OneCycleLR",
    max_lr=optimizer["lr"],
    pct_start=0.04,
    anneal_strategy="cos",
    div_factor=10.0,
    final_div_factor=100.0,
)

# dataset settings
dataset_type = "rellisDataset"
data_root = "/data/rellis_3d"
ignore_index = 2 # sky
names = [
    'vegetation',
    'terrain',
    'sky',
    'construction',
    'vehicle',
    'road',
    'object',
    'void',
    'sign',
    'person',
    'water',
    'animal'
]

data = dict(
    num_classes=12,
    ignore_index=ignore_index,
    names=names,
    train=dict(
        type=dataset_type,
        split= "train",
        # split="train",
        data_root=data_root,
        transform=[
            # dict(type="RandomDropout", dropout_ratio=0.2, dropout_application_ratio=0.2),
            # dict(type="RandomRotateTargetAngle", angle=(1/2, 1, 3/2), center=[0, 0, 0], axis="z", p=0.75),
            dict(type="RandomRotate", angle=[-1, 1], axis="z", center=[0, 0, 0], p=0.5),
            # dict(type="RandomRotate", angle=[-1/6, 1/6], axis="x", p=0.5),
            # dict(type="RandomRotate", angle=[-1/6, 1/6], axis="y", p=0.5),
            dict(type="RandomScale", scale=[0.9, 1.1]),
            # dict(type="RandomShift", shift=[0.2, 0.2, 0.2]),
            dict(type="RandomFlip", p=0.5),
            dict(type="RandomJitter", sigma=0.005, clip=0.02),
            # dict(type="ElasticDistortion", distortion_params=[[0.2, 0.4], [0.8, 1.6]]),
            dict(
                type="GridSample",
                grid_size=0.05, # 0.05
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            # dict(type="SphereCrop", point_max=300000, mode="random"),
            # dict(type="CenterShift", apply_z=False),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
        ignore_index=ignore_index,
    ),
    val=dict(
        type=dataset_type,
        split="val",
        # split="val",
        data_root=data_root,
        transform=[
            # dict(type="PointClip", point_cloud_range=(-51.2, -51.2, -4, 51.2, 51.2, 2.4)),
            dict(
                type="GridSample",
                grid_size=0.05, # 0.05
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            # dict(type="SphereCrop", point_max=300000, mode='center'),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
        ignore_index=ignore_index,
    ),
    test=dict(
        type=dataset_type,
        split="val",
        # split="test",
        data_root=data_root,
        transform=[
            dict(type="Copy", keys_dict={"segment": "origin_segment"}),
            dict(
                type="GridSample",
                grid_size=0.025,
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_inverse=True,
            ),
            # dict(type="SphereCrop", point_max=300000, mode='center'),
        ],
        test_mode=True,
        test_cfg=dict(
            voxelize=dict(
                type="GridSample",
                grid_size=0.05, # 0.05
                hash_type="fnv",
                mode="test",
                return_grid_coord=True,
                keys=("coord", "strength"),
            ),
            crop=None,
            post_transform=[
                dict(type="ToTensor"),
                dict(
                    type="Collect",
                    keys=("coord", "grid_coord", "index"),
                    feat_keys=("coord", "strength"),
                ),
            ],
            aug_transform=[
                [dict(type="RandomRotateTargetAngle", angle=[0], axis="z", center=[0, 0, 0], p=1)],
            ],
        ),
        ignore_index=ignore_index,
    ),
)

``` 

validation result

<img width="427" height="218" alt="Image" src="https://github.com/user-attachments/assets/f2a78ed6-6777-4260-b150-c029a14acc80" />

test result

<img width="433" height="250" alt="Image" src="https://github.com/user-attachments/assets/03313dd1-7e4f-4f4c-b3aa-7117114bf8e2" />

I would be very grateful if you could help me analyze the cause.

Thanks

**Comment by ATh0ft** [comment]
Hi 
It seem that you are using a different grid size in the test transforms. Secondly, if the model uses batch norm or dropout, these would be disabled when running the testing as you have test_mode=True for the testing but not the validation. 
I hope these pointers are of help :)

**Comment by Gofinge** [comment]
Hey, this is a hybrid way to do our precise evaluation and mapping evaluation. Point Cloud is initially grid sampled with 1/2 * default grid size and then separated into multiple fragments with 1 * default grid size. We do inference on point cloud with 1 * default grad size and ensemble these prediction to middle-level of point cloud with 1/2 * default grid size and mapping it back to orginal scale.
